### PR TITLE
test: Edge cases - Token Service Tests Part 3

### LINKED ISSUE #19757: Edge cases - Token Service Tests

### PR DESCRIPTION
**Pull request by vtronkov**
**Description**:
* Part 3 of 3 to add token edge cases wrapped in an atomic batch
* Fix `HapiGetTokenNftInfo`'s `hasValidCreationTime` to work with batch transactions

**Related issue(s)**:

Fixes #19757

### CODE REVIEW

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/batch/AtomicTokenUpdateSpecs.java [hunk 1]
```diff
@@ -0,0 +1,738 @@
...
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Tag;
+
+// This test cases are direct copies of TokenCreateSpecs. The difference here is that
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/batch/AtomicTokenUpdateSpecs.java]:
```suggestion
// This test cases are direct copies of TokenUpdateSpecs. The difference here is that
```
?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenNftInfo.java [hunk 2]
```diff
@@ -112,8 +118,16 @@ protected void assertExpectationsGiven(HapiSpec spec) throws Throwable {
 
         expectedMetadata.ifPresent(bytes -> assertEquals(bytes, actualInfo.getMetadata(), "Wrong metadata!"));
 
+        // If the NFT is created inside a batch then we increase the nanoseconds by 1 since the exact creation time
+        // is taken from the batch itself.
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenNftInfo.java]: Is this expected, or a bug?

**Review by mhess-swl — CHANGES_REQUESTED** [review]
It looks like all the files in this PR have `This test cases are direct copies of TokenCreateSpecs.` Test cases seem copied fine, just the names of the tests they were copied from weren't changed.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenNftInfo.java (hunk 2 above)
Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenNftInfo.java]: I think it's expected. The first nanosecond is saved for the batch, and each inner transaction is incremented by one. This is my understatement of how it works, and that's why I've extended the verb to support that.

Comment (mhess-swl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenNftInfo.java]: Is the transaction time guaranteed to increase by 1 in all cases (e.g. could it be +2, +3, something else)? Dispatch uses a nontrivial algorithm to order all the child/inner transactions

Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenNftInfo.java]: I would think so, but I will ask the others for their input

Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenNftInfo.java]: You are right. @JivkoKelchev found a problem with this approach and came up with a solution. He will open a PR shortly. I will revert my changes from this PR and wait for his to be merged. Thank you both!

Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/queries/token/HapiGetTokenNftInfo.java]: I also deleted `AtomicUniqueTokenManagementSpecs` test because it's moved to Jivko's PR to verify his fix - https://github.com/hiero-ledger/hiero-consensus-node/pull/19866
